RecoVertex/BeamSpotProducer/python/BeamSpot_cff.py

```diff
@@ -9,3 +9,8 @@
 _offlineBeamSpotTask_gpu = offlineBeamSpotTask.copy()
 _offlineBeamSpotTask_gpu.add(offlineBeamSpotToCUDA)
 gpu.toReplaceWith(offlineBeamSpotTask, _offlineBeamSpotTask_gpu)
+
+def _addCUDAServices(theProcess):
+     theProcess.load("HeterogeneousCore.CUDAServices.CUDAService_cfi")
+
+modifyRecoVertexBeamSpotProducerBeamSpotAddCUDAService_ = gpu.makeProcessModifier( _addCUDAServices )
```

Comment:
It was sort of agreed in https://github.com/cms-sw/cmssw/issues/28575 to place this in `Configuration/StandardSequences/Services_cff`. Unless we want to place this in every future cfi/cff that introduces a CUDA module.

Reply:
Indeed a central place would be better.

By the way, how do we deal with this for the GCC 10 builds ?
The `CUDAService` does not build there (though maybe in principle it could), so there is no `CUDAService_cfi`.

Reply:
> By the way, how do we deal with this for the GCC 10 builds ?
> The `CUDAService` does not build there (though maybe in principle it could), so there is no `CUDAService_cfi`.

It will indeed continue to fail in GCC 10 on workflows enabling `gpu` modifier (better than all workflows but not fully satisfactory). Once we agree with the immediate-term solution in #31261, the same pattern could be applied to `CUDAService` as well.

On the other hand, I'd think the `gpu` workflows would not make much sense on GCC 10 anyway, so should we look into disabling them in GCC 10 IBs? (in which case we could leave `CUDAService` untouched for now)

Reply:
I think a better solution for the `CUDAService` could be to try and build it also on GCC 10.

With
```diff
diff --git a/HeterogeneousCore/CUDAServices/BuildFile.xml b/HeterogeneousCore/CUDAServices/BuildFile.xml
index e063a0dd4e7..945d272ba85 100644
--- a/HeterogeneousCore/CUDAServices/BuildFile.xml
+++ b/HeterogeneousCore/CUDAServices/BuildFile.xml
@@ -1,4 +1,4 @@
-<iftool name="cuda-gcc-support">
+<iftool name="cuda">
   <use name="FWCore/Framework"/>
   <use name="FWCore/ServiceRegistry"/>
   <use name="FWCore/ParameterSet"/>
diff --git a/HeterogeneousCore/CUDAServices/plugins/BuildFile.xml b/HeterogeneousCore/CUDAServices/plugins/BuildFile.xml
index 73a760fa117..0928247e400 100644
--- a/HeterogeneousCore/CUDAServices/plugins/BuildFile.xml
+++ b/HeterogeneousCore/CUDAServices/plugins/BuildFile.xml
@@ -1,4 +1,4 @@
-<iftool name="cuda-gcc-support">
+<iftool name="cuda">
   <use name="cuda"/>
   <use name="DataFormats/Common"/>
   <use name="DataFormats/Provenance"/>
@@ -11,6 +11,7 @@
   <use name="FWCore/ServiceRegistry"/>
   <use name="FWCore/Utilities"/>
   <use name="HeterogeneousCore/CUDAServices"/>
+
   <library file="*.cc" name="HeterogeneousCoreCUDAServicesPlugins">
     <flags EDM_PLUGIN="1"/>
   </library>
diff --git a/HeterogeneousCore/CUDAUtilities/BuildFile.xml b/HeterogeneousCore/CUDAUtilities/BuildFile.xml
index fc2752ff845..e7193e1ccfd 100644
--- a/HeterogeneousCore/CUDAUtilities/BuildFile.xml
+++ b/HeterogeneousCore/CUDAUtilities/BuildFile.xml
@@ -1,4 +1,4 @@
-<iftool name="cuda-gcc-support">
+<iftool name="cuda">
   <use name="cuda"/>
   <use name="eigen"/>
   <use name="FWCore/Utilities"/>
```
the `CUDAService` can be built and loaded.

Reply:
The idea is that `<iftool name="cuda">` tells if we have the CUDA libraries, while `<iftool name="cuda-gcc-support">` tells if we can compile the device code.

Reply:
See https://github.com/cms-sw/cmssw/pull/31334 .